server/schedule/operator/create_operator.go

```diff
@@ -338,112 +123,28 @@ func CreateMergeRegionOperator(desc string, cluster Cluster, source *core.Region
+	return true
 }
 
 // CreateScatterRegionOperator creates an operator that scatters the specified region.
-func CreateScatterRegionOperator(desc string, cluster Cluster, origin *core.RegionInfo, replacedPeers, targetPeers []*metapb.Peer) *Operator {
-	// Randomly pick a leader
-	i := rand.Intn(len(targetPeers))
-	targetLeaderPeer := targetPeers[i]
-	originLeaderStoreID := origin.GetLeader().GetStoreId()
-
-	originStoreIDs := origin.GetStoreIds()
-	steps := make([]OpStep, 0, len(targetPeers)*3+1)
-	// deferSteps will append to the end of the steps
-	deferSteps := make([]OpStep, 0, 5)
-	var kind OpKind
-	sameLeader := targetLeaderPeer.GetStoreId() == originLeaderStoreID
-	// No need to do anything
-	if sameLeader {
-		isSame := true
-		for _, peer := range targetPeers {
-			if _, ok := originStoreIDs[peer.GetStoreId()]; !ok {
-				isSame = false
-				break
-			}
-		}
-		if isSame {
-			return nil
-		}
-	}
-
-	// Creates the first step
-	if _, ok := originStoreIDs[targetLeaderPeer.GetStoreId()]; !ok {
-		st := CreateAddLightPeerSteps(targetLeaderPeer.GetStoreId(), targetLeaderPeer.GetId())
-		steps = append(steps, st...)
-		// Do not transfer leader to the newly added peer
-		// Ref: https://github.com/tikv/tikv/issues/3819
-		deferSteps = append(deferSteps, TransferLeader{FromStore: originLeaderStoreID, ToStore: targetLeaderPeer.GetStoreId()})
-		deferSteps = append(deferSteps, RemovePeer{FromStore: replacedPeers[i].GetStoreId()})
-		kind |= OpLeader
-		kind |= OpRegion
-	} else {
-		if !sameLeader {
-			steps = append(steps, TransferLeader{FromStore: originLeaderStoreID, ToStore: targetLeaderPeer.GetStoreId()})
-			kind |= OpLeader
-		}
-	}
-
-	// For the other steps
-	for j, peer := range targetPeers {
-		if peer.GetId() == targetLeaderPeer.GetId() {
-			continue
-		}
-		if _, ok := originStoreIDs[peer.GetStoreId()]; ok {
-			continue
-		}
-		if replacedPeers[j].GetStoreId() == originLeaderStoreID {
-			st := CreateAddLightPeerSteps(peer.GetStoreId(), peer.GetId())
-			st = append(st, RemovePeer{FromStore: replacedPeers[j].GetStoreId()})
-			deferSteps = append(deferSteps, st...)
-			kind |= OpRegion | OpLeader
-			continue
-		}
-		st := CreateAddLightPeerSteps(peer.GetStoreId(), peer.GetId())
-		steps = append(steps, st...)
-		steps = append(steps, RemovePeer{FromStore: replacedPeers[j].GetStoreId()})
-		kind |= OpRegion
-	}
-
-	steps = append(steps, deferSteps...)
-
-	targetStores := make([]uint64, len(targetPeers))
-	for i := range targetPeers {
-		targetStores[i] = targetPeers[i].GetStoreId()
-	}
-	sort.Sort(u64Slice(targetStores))
-	brief := fmt.Sprintf("scatter region: stores %v to %v", u64Set(origin.GetStoreIds()), targetStores)
-	op := NewOperator(desc, brief, origin.GetID(), origin.GetRegionEpoch(), kind, steps...)
-	return op
+func CreateScatterRegionOperator(desc string, cluster Cluster, origin *core.RegionInfo, targetPeers map[uint64]*metapb.Peer) (*Operator, error) {
+	return NewBuilder(desc, cluster, origin).
```

Comment:
`ScatterRegion` will randomly peer a target leader, Is this behavior consistent?

Reply:
No. I'll fix it.

Reply:
fixed.